libpod/container_internal_linux.go

```diff
@@ -668,11 +669,30 @@ func (c *Container) setupSystemd(mounts []spec.Mount, g generate.Generator) erro
 		}
 		g.AddMount(systemdMnt)
 	} else {
+		hostMounts, err := mount.GetMounts()
+		if err != nil {
+			return err
+		}
+
+		var hostMount *mount.Info
+		for _, m := range hostMounts {
+			if m != nil && m.Mountpoint == "/sys/fs/cgroup/systemd" {
+				hostMount = m
+				break
+			}
+		}
+
+		if hostMount == nil {
+			return errors.Wrapf(define.ErrInternal, "/sys/fs/cgroup/systemd is not mounted on the host")
+		}
+
```

Comment:
this is a bit expensive.

Could we just use `statfs("/sys/fs/cgroup/systemd")` and read `f_flags` from there?